helix-rest/src/main/java/org/apache/helix/rest/common/HelixDataAccessorWrapper.java

```diff
@@ -101,7 +118,7 @@ public HelixDataAccessorWrapper(ZKHelixDataAccessor dataAccessor) {
   }
 
   private Map<String, Boolean> getPartitionsHealth(String instance, ZNRecord partitionHealthRecord,
-      RESTConfig restConfig, Map<String, String> customPayLoads) {
+      RESTConfig restConfig, Map<String, String> customPayLoads, boolean requireFullRead) {
```

Comment:
I suggest we use something like `onlyUseZkAsSource` or something as the boolean flag, and use the same name all throughout since having `skipZkRead` and `requireFullRead` here and there could be confusing down the road.

Reply:
I understand your concern. But it will be confusing to use one flag because they are completely in different purpose:
1. Skip the zk access for reading partition health.
2. Because of the contract we made for customized API, if no entry as body for "PARTITION", they will return all the partition health status. If they have it but partition list is empty, no result will be return. Here' when we only do partition health retrieving, we dont know what are the partitions on the instance. Also this function I would say is a separate function and logically not connected with "skipZKRead"

I can add some comments in the function and rename the function. But I do not think make this name to be "onlyUseZkAsSource" is a good choice. There are no logic related to ZK for this private function.